### DISCUSSION TITLE
C++11 compatibility

### DISCUSSION THREAD

**Discussion by thegwydd** [daniele77/cli#169] [Ideas / Feature requests]
Really nice library!

I would like to see this library as C++ 11 library.

Many embedded platform toolchains are hard to update (no yocto, no build root, no sdk generation tools) and it's nearly impossibile/totally inconvenient to update compiler. 
I've seen many platforms stuck with C++11 and it would be nice to use this library on such platforms.

Furthermore I've seen the only used features of C++14 are usage of auto on lambdas parameters, usage of auto on function return values and usage of make_unique.

Last year I successfully ported v1.2.1 to C++11 and compiled with gcc 4.9.2 and now I've seen that v2.0.2 is really similar.
Would you accept and then maintain a C++11 conversion?

**Comment by daniele77** [comment]
Thank you.
I know that a lot of embedded software is stuck with C++11 (or previous).
Please send a PR.
